moveit_core/collision_detection_bullet/src/collision_env_bullet.cpp

```diff
@@ -89,6 +89,9 @@ CollisionEnvBullet::CollisionEnvBullet(const CollisionEnvBullet& other, const Wo
   {
     addLinkAsCollisionObject(link.second);
   }
+
+  // get notifications any objects already in the new world
+  getWorld()->notifyObserverAllObjects(observer_handle_, World::CREATE);
```

Comment:
That's quite involved. I'd have to invest quite a bit of time to see whether this is actually correct.
A test/tests to illustrate the different behavior would be appreciated here.

The bullet plugin was based on the FCL checker.
Can you explain why [the same constructor for FCL works without this additional line?](https://github.com/ros-planning/moveit/blob/master/moveit_core/collision_detection_fcl/src/collision_env_fcl.cpp#L152-L163)

Reply:
I believe FCL updates the world objects here https://github.com/ros-planning/moveit/blob/master/moveit_core/collision_detection_fcl/src/collision_env_fcl.cpp#L157-L159 which is why it does not call the notifier.

I will work on some tests and ping you after I get them to work.